rs/crypto/src/vetkd/mod.rs

```diff
@@ -0,0 +1,560 @@
+                .map(|(node_id, node_index, clib_share)| {
+                    let node_public_key = lazily_calculated_public_key_from_store(
+                        lockable_threshold_sig_data_store,
+                        threshold_sig_csp_client,
+                        &args.ni_dkg_id,
+                        *node_id,
+                    )
+                    .map_err(|e| {
+                        VetKdKeyShareCombinationError::IndividualPublicKeyComputationError(e)
+                    })?;
+                    let node_public_key_g2affine = match node_public_key {
+                        CspThresholdSigPublicKey::ThresBls12_381(public_key_bytes) => {
+                            ///////////////////////////////////////////////////////////////
+                            // TODO: Can/should we use G2Affine::deserialize_unchecked here
+                            G2Affine::deserialize(&public_key_bytes.0)
+                            .map_err(|_: PairingInvalidPoint| VetKdKeyShareCombinationError::InternalError(
+                                format!("individual public key of node with ID {node_id} in threshold sig data store")
+                            ))
+                        }
+                    }?;
+                    Ok((*node_index, node_public_key_g2affine, clib_share.clone()))
+                })
+                .collect::<Result<_, _>>()?;
+
+            ic_crypto_internal_bls12_381_vetkd::EncryptedKey::combine_valid_shares(
+                &clib_shares_for_combine_valid[..],
+                reconstruction_threshold,
+                &master_public_key,
+                &transport_public_key,
+                &derivation_path,
+                &args.derivation_id,
+            )
+            .map_err(|e| {
+                VetKdKeyShareCombinationError::CombinationError(format!(
+                    "failed to combine the valid encrypted vetKD key shares: {e:?}"
+                ))
+            })
+        },
+        Err(other_error) => {
+            Err(VetKdKeyShareCombinationError::CombinationError(format!(
+                "failed to combine the valid encrypted vetKD key shares: {other_error:?}"
+            )))
+        }
+    }
+    .map(|encrypted_key| VetKdEncryptedKey {
+        encrypted_key: encrypted_key.serialize().to_vec(),
+    })
+}
+
+fn share_to_clib_share(
+    share: &ic_types::crypto::vetkd::VetKdEncryptedKeyShare,
+) -> Result<ic_crypto_internal_bls12_381_vetkd::EncryptedKeyShare, VetKdKeyShareCombinationError> {
+    let encrypted_key_share_192_bytes = {
+        if share.encrypted_key_share.0.len() != EncryptedKeyShare::BYTES {
+            return Err(VetKdKeyShareCombinationError::InvalidArgumentEncryptedKeyShare);
+        }
+        let mut encrypted_key = [0u8; EncryptedKeyShare::BYTES];
+        encrypted_key.copy_from_slice(&share.encrypted_key_share.0);
+        encrypted_key
+    };
+    ic_crypto_internal_bls12_381_vetkd::EncryptedKeyShare::deserialize(
```

Comment:
This function and `verify_encrypted_key_internal` would be a bit simpler if `EncryptedKeyShare::deserialize` and `EncryptedKey::deserialize` accepted a slice and just returned an error if the slice length was not expected. Since we don't distinguish between the two error cases (wrong length vs invalid point) there would be no loss in terms of reporting